### PR TITLE
EVG-7773: banners for error, success, warning and info messages

### PR DESCRIPTION
**Pull request by tgrander**
banners

These banners will be displayed at the top of the page and at the component level, such as in the tasks tab on the patch page. 

### Summary
- Singular `<Banner/>` components - will not be used alone / will only be used via `<Banners/>`
- `<Banners/>` component - shows stacked list of banners
- Banners context - with this context, the banners can be used in any part of the app and a banner can be dispatched by any child component by using the `useBannerDispatchContext` hook

An example of how to use `<Banners/>` is demonstrated in Storybook. The code can be viewed in the banner storybook file. 

### Next steps
Use banners component and context to render error and success messages across the entire app.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,8 +51,7 @@ Follow these directions to enable query linting during local development so your
     "REACT_APP_GQL_URL": "http://localhost:9090/graphql/query",
     "REACT_APP_API_URL": "http://localhost:3000/api",
     "REACT_APP_UI_URL": "http://localhost:9090",
-    "REACT_APP_SPRUCE_URL": "http://localhost:3000",
-    "REACT_APP_GQL_COOKIE": "cookie-from-REACT_APP_GQL_URL"
+    "REACT_APP_SPRUCE_URL": "http://localhost:3000"
```
Comment (tgrander) [inline README.md:54]: i tried so many different things to get this to go away. After this PR I am going to delete my fork and re-fork from evergreen/spruce

File: src/context/banners.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  clearAll: () => void;
+}
+
+const BannerDispatchContext = React.createContext<DispatchBanner | null>(null);
+const BannerStateContext = React.createContext<BannerObj[] | null>(null);
+
+export const BannerContextProvider: React.FC = ({ children }) => {
+  const [banners, setBanners] = useState<BannerObj[]>([]);
+
+  const addBanner = (type: BannerTypeKeys, message: string) => {
```
Comment (khelif96) [inline src/context/banners.tsx:26]: Lara had some ideas about the ordering of banner priority. She wants the warning banner to always have priority. She mentioned placing it in the wiki but i dont think that was made yet.

File: README.md (hunk 1 above)
Comment (SupaJoon) [reply README.md:54]: we should reset the changes for this file

File: src/context/banners.tsx (hunk 2 above)
Comment (tgrander) [reply src/context/banners.tsx:26]: done

File: package.json [hunk 3]
```diff
@@ -41,6 +41,7 @@
     "react-dom": "16.12.0",
     "react-router-dom": "5.1.2",
     "react-scripts": "3.4.0",
+    "uuid": "^8.0.0",
```
[the remark this answers is not in the document]
Comment (tgrander) [reply package.json]: done

File: src/stories/1-Banner.stories.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
+import React from "react";
```
[the remark this answers is not in the document]
Comment (tgrander) [reply src/stories/1-Banner.stories.js:1]: :)

File: src/components/Banners/Banner.tsx [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+  children,
+  type,
+  removeBanner,
+  id,
+}) => {
+  const onClickX = () => removeBanner(id);
+  return (
+    <Wrapper type={type}>
+      <StyledBody type={type}>{children}</StyledBody>
+      {(type === "success" || type === "info") && (
```
Comment (SupaJoon) [inline src/components/Banners/Banner.tsx:26]: why not allow warning banners to be removable?

File: src/stories/1-Banner.stories.js [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+        <Button onClick={() => banner.error("I am a failure")}>Error</Button>
+        <Button
+          onClick={() =>
+            banner.warning("Not a failure but not a success either")
+          }
+        >
+          Warning
+        </Button>
+        <Button
+          onClick={() => banner.info("I am a harmless informational banner")}
```
Comment (SupaJoon) [inline src/stories/1-Banner.stories.js:50]: cool useage

File: src/components/Banners/Banner.tsx (hunk 5 above)
Comment (tgrander) [reply src/components/Banners/Banner.tsx:26]: because designs

File: src/stories/1-Banner.stories.js (hunk 6 above)
Comment (tgrander) [reply src/stories/1-Banner.stories.js:50]: 💯

**Pull request merged**